lix/source-code/client/src/browser-auth.ts

```diff
@@ -0,0 +1,98 @@
+import { publicEnv } from "@inlang/env-variables"
+
+const gitHubProxyUrl = publicEnv.PUBLIC_GIT_PROXY_BASE_URL + "/github-proxy/"
+
+// the authUrl for the oauth app with minimal scopes https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps
+// oauth apps can only require public (public_repo) or private repos but no further repo or org selection
+// email is required to commit with identity of who committed
+const authUrl = `https://github.com/login/oauth/authorize?client_id=${publicEnv.PUBLIC_GITHUB_APP_CLIENT_ID}&scope=repo,user:email`
+
+/**
+ * Login user in new window.
+ *
+ * Read https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#1-request-a-users-github-identity
+ * works only in browsers for now, but other methods should be supported in future
+ */
+// TODO: later use url with default instead of env var: args: { url?: string }
+export function login() {
+	window.open(authUrl, "_blank")
+}
+export function logout() {
+	return fetch(`${publicEnv.PUBLIC_GIT_PROXY_BASE_URL}/services/auth/sign-out`, {
+		method: "POST",
+		credentials: "include",
+	})
+}
+
+interface Email {
```

Comment:
@janfjohannes does this belong in here (also not exported) or on a more global level? could imagine that this could probably be used by multiple modules.

Reply:
On that note, the rest of the codebase uses `type`. Probably wise to choose `type` instead of `interface`.

Reply:
@felixhaeberle   @samuelstroschein  i changed from interface to type but i dont want to expose it as this is typing according to github logic and i would prefer to only expose and reuse types more if we created them specifically for lix, tooling can still use it with return type inference.